[KB] Put the Drawer in MiniMode on small screens (#851)
telerik/blazor-docs: knowledge-base/drawer-small-screen.md

---
title: Put the Drawer in MiniMode on small screens
description: How to put the Drawer in MiniMode on small screens.
type: how-to
page_title: Put the Drawer in MiniMode on small screens
slug: drawer-kb-minimode-smallscreens
position:
tags:
res_type: kb

> ---

Reviewer: ticketid metadata missing

## Environment
<table>
	<tbody>
		<tr>
			<td>Product</td>
			<td>Drawer for Blazor</td>
		</tr>
	</tbody>
</table>

## Description

I would like to put the Drawer in [MiniMode]({%slug drawer-mini-mode%}) on small screens. This behavior would be great for mobile devices. 

## Solution

> 1. The Blazor application has to be aware of the current browser width and you can use the [TelerikMediaQuery]({%slug mediaquery-overview%}) to achieve the desired behavior.

Reviewer: ```suggestion
1. The Blazor application has to be aware of the current browser width. One way to achieve this is to use the [TelerikMediaQuery component]({%slug mediaquery-overview%}).
```

> 2. In the OnChange event handler you can put the Drawer in MiniMode.

Reviewer: Link to the event docs.

```suggestion
2. Use the MediaQuery `OnChange` event to put the Drawer in MiniMode.
```

>caption Put the Drawer in MiniMode on small screens

````CSHTML

> @* This example assumes that a max-width of 767px would be a suitable match for a small screen size. You can adjust that value as needed in your application.

Reviewer: ```suggestion
@* This example assumes a max-width of 767px as a suitable match for a small screen size. You can adjust that value in your application.
```

> To see the behavior, you can resize the browsers window*@

Reviewer: ```suggestion
To see the behavior, resize the browser window *@
```

<TelerikMediaQuery Media="(max-width: 767px)" 
                   OnChange="@((bool doesMatch) => OnChangeSmallScreenHandler(doesMatch))">
</TelerikMediaQuery>

<TelerikButton OnClick="@(() => DrawerRef.ToggleAsync())" Icon="menu">Toggle drawer</TelerikButton>

<TelerikDrawer Data="@Data"
               MiniMode="@isSmallScreen"
               Mode="@DrawerMode.Push"
               @ref="@DrawerRef">
</TelerikDrawer>

@code {
    private bool isSmallScreen { get; set; }

    private void OnChangeSmallScreenHandler(bool doesMatch)
    {
        //Triggers MiniMode on Small screens
        isSmallScreen = doesMatch;

>         //Alternatively, you can use the CollapseAsync() method

Reviewer: Why not have the code here, even if its commented out?

---
title: Put the Drawer in MiniMode on small screens
description: How to put the Drawer in MiniMode on small screens.
type: how-to
page_title: Put the Drawer in MiniMode on small screens
slug: drawer-kb-minimode-smallscreens
position:
tags:
res_type: kb
---

## Environment
<table>
	<tbody>
		<tr>
			<td>Product</td>
			<td>Drawer for Blazor</td>
		</tr>
	</tbody>
</table>

## Description

I would like to put the Drawer in [MiniMode]({%slug drawer-mini-mode%}) on small screens. This behavior would be great for mobile devices. 

## Solution

1. The Blazor application has to be aware of the current browser width. One way to achieve this is to use the [TelerikMediaQuery component]({%slug mediaquery-overview%}).

> 2. Use the MedidQuery `[OnChange event]({%slug mediaquery-events%})` to put the Drawer in MiniMode.

Reviewer: ```suggestion
2. Use the MediaQuery `[OnChange event]({%slug mediaquery-events%})` to put the Drawer in MiniMode.
```
